### DISCUSSION TITLE
How to open chat in right split window by CopilotChat.

### DISCUSSION THREAD

**Discussion by yfujita-skgcat** [CopilotC-Nvim/CopilotChat.nvim#124] [Q&A] [answered]
Is there any simple way to display the chat from the CopilotChat command in a split window on the right side?

I read "rplugin/python3/CopilotChat/mypynvim/core/buffer.py" and found self.nvim.api.command(f"vsplit | buffer {self.buf.handle}").
I inserted "wincmd l" as follows and succeeded in open chat in right side.
```
    def vsplit(self, opts: dict[str, Any] = {}):
        # self.nvim.api.command(f"vsplit | buffer {self.buf.handle}")
        self.nvim.api.command(f"vsplit | wincmd l | buffer {self.buf.handle}")
        winnr = self.nvim.api.get_current_win()
        for opt, val in opts.items():
            self.nvim.api.win_set_option(winnr, opt, val)
```

Is there any other way than modifying the source code?

**Comment by deathbeam** [comment, marked as answer]
set splitright if you want to have it globally (which is imo the most sane default)
and set splitbelow for horizontal split

**Reply by Infonautica** [reply]
For the future visitors, this is the change that makes chat to open on the right:
```lua
vim.opt.splitright = true
```

**Reply by funmathis** [reply]
I know you asked about doing without modifying the code, but it is literally a one character change. Change this
```
function! CopilotChat()
  call LoadConfig() 
  call ConfirmSignin()
  vsplit
...
```
to this
```
function! CopilotChat()
  call LoadConfig() 
  call ConfirmSignin()
  split
...
```
in copilot_chat.vim
#######################

Whoops, that is for this repo: https://github.com/DanBradbury/copilot-chat.vim for vim. I imagine there is something similar here
